src/entity/entity.rs

```diff
@@ -0,0 +1,91 @@
+use crate::graphics::entity::EntityCube;
+use crate::entity::humanoid;
+use crate::vector::Vector3;
+
+
+#[derive(Debug, PartialEq)]
+/// Enum for the different types of entity
+pub enum EntityKind {
+    Player
+}
+
+/// Contain the data of an entity
+pub struct Entity {
+    id: usize,
+    entity_type: EntityKind,
+    position: Vector3,
+    orientation: [f32;2],
+}
+
+impl Entity {
+    pub fn new(id: usize, entity_type: EntityKind, position: Vector3, orientation: [f32;2]) -> Self {
+        Self { 
+            id,
+            entity_type,
+            position,
+            orientation,
+        }
+    }
+    
+    pub fn set_position(&mut self, position: Vector3) {
+        self.position = position;
+    }
+    
+    pub fn set_orientation(&mut self, orientation: [f32;2]) {
+        self.orientation = orientation;
+    }
+    
+    pub fn position(&self) -> &Vector3 {
+        &self.position
+    }
+    
+    pub fn orientation(&self) -> [f32; 2] {
+        self.orientation
+    }
+        
+    pub fn id(&self) -> usize {
+        self.id
+    }
+        
+    pub fn entity_type(&self) -> &EntityKind {
+        &self.entity_type
+    }
+
+    /// Draw the entity and return a Vec of EntityCube
+    pub fn draw_entity(&mut self) -> Vec<EntityCube> {
```

Comment:
Ok, to be consistent, the name _draw_ is not very good. 

"Draw" is a verb so it emphasis that the method will actually draw something. 

Please rename the "draw" into "get_opengl_entities".

We will also do soon a refactoring of the names for RectAttr and these shit names that I had came up with👍